### DISCUSSION TITLE
RecommendationController using SNAP

### DISCUSSION THREAD

**Discussion by tina86-hub** [searchspring/snap#939] [Q&A] [answered]
Hi,
 I can see the results when using API link for recommended products -
https://api.searchspring.net/api/recommend/trending.json?siteId=xxxxx&limit=16.

My question is how to integrate it with SNAP and display these products on search results page. Our config has
	instantiators: {
		recommendation: {
			components: {
				Default: async () => {
					return (await import('./components/Recommendations/Recs')).Recs;
				},
			},
			config: {
				branch: 'snapdev',
				tag: 'trending',
				id: 'recommend',
			},
		},
	}

and then in results.jsx we have code as 
	<p> No results found. Showing top popular products you might want to consider...</p>
	<Recs store={store} controller={controller}  title={'Popular Products'}/>

BUT it's not returning/displaying any products. 

Thanks

**Comment by korgon** [comment, marked as answer]
Hello @tina86-hub,

Sorry for the late response, your question has gone un-noticed until now - we will make an effort to pick these up in a more timely manner.

With regards to your question, if you are wanting to simply get the most popular queries (trending) and display them on a no-results page, you won't be needing to utilize the recommendations component, nor the instantiator. The instantiator is used for targeting and injecting into recommendation script blocks found on the page (see: [recommendation integration instructions](https://searchspring.github.io/snap/#/integration-recommendations)).

The autocomplete controller typically handles these types of suggested queries (trending or otherwise) and so it is equipped with proper store to hold this data. Assuming you have created an autocomplete controller in the root config, you can "connect" this controller with the search controller in order to access it within the `results.jsx` component.

Example code for connecting the controllers inside `index.js`:
```
// main snap config
const config = {
	...
}

const snap = new Snap(config);

/* link the search and autocomplete controllers */
snap.getControllers('search', 'autocomplete').then(([search, autocomplete]) => {
	search.autocompleteController = autocomplete;
});
```

Example using the connected controller inside of a component that expects the search controller as a prop (eg: `results.jsx`):
```
		...

		const acTrendingTerms = controller.autocompleteController?.store?.trending;

		return (
			<div className="ss__no-results">

				<div>
					<p>Popular searches:</p>
					<ul>
						{
							acTrendingTerms.map((term) => <li><a {...term.url.link}>{ term.value }</a></li>)
						}
					</ul>
				</div>

				...
			)

		...
```

**Comment by tina86-hub** [comment]
Thanks for the reply @korgon.
I have another question - how can we set the infinite scroll (with "Load More" button) to stay on the same location when loading the next page rather than scrolling up to the top. I have tried the properties such as restorePosition.enabled, but may be using them wrong.

                            `id: 'search',
					plugins: [[plugin]],
					globals: {
						 filters: backgroundFilters,
						 pagination: {
								 pageSize: 40
						 },
						 facet: {
			 					 limit: 2,
			 					collapsed: true
			 			 },

				 },
					settings: {
								infinite: {
										backfill: 5
								}

					},`

Thanks

**Reply by korgon** [reply]
Scrolling is mostly handled within middleware - in the project code. In our standard project scaffolding this is done in the 'restorePosition' event:

https://github.com/searchspring/snapfu-scaffold-preact/blob/production/src/scripts/plugin.js#L5

If you have similar middleware (yours may be happening on `afterStore`), you can simply remove scrolling to the top there, or put additional logic around it to scroll only when you want (for example all filters / sorts will reset pagination to page 1).

Hope that helps!

**Reply by tina86-hub** [reply]
Thanks @korgon. This was helpful :)
